paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/kafka/KafkaSyncTableAction.java

```diff
@@ -0,0 +1,354 @@
+import org.apache.paimon.catalog.Identifier;
+import org.apache.paimon.flink.FlinkConnectorOptions;
+import org.apache.paimon.flink.action.Action;
+import org.apache.paimon.flink.action.cdc.ComputedColumn;
+import org.apache.paimon.flink.action.cdc.kafka.canal.CanalJsonEventParser;
+import org.apache.paimon.flink.sink.cdc.EventParser;
+import org.apache.paimon.flink.sink.cdc.FlinkCdcSyncTableSinkBuilder;
+import org.apache.paimon.options.CatalogOptions;
+import org.apache.paimon.options.Options;
+import org.apache.paimon.schema.Schema;
+import org.apache.paimon.table.FileStoreTable;
+
+import org.apache.flink.api.common.eventtime.WatermarkStrategy;
+import org.apache.flink.api.java.tuple.Tuple3;
+import org.apache.flink.api.java.utils.MultipleParameterTool;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.connector.kafka.source.KafkaSource;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.connectors.kafka.table.KafkaConnectorOptions;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.apache.paimon.utils.Preconditions.checkArgument;
+
+/**
+ * An {@link Action} which synchronize one kafka topic into one Paimon table.
+ *
+ * <p>This topic must be from canal-json format.
+ *
+ * <p>You should specify Kafka source topic in {@code kafkaConfig}. See <a
+ * href="https://nightlies.apache.org/flink/flink-docs-release-1.16/zh/docs/connectors/table/kafka/">document
+ * of flink-connectors</a> for detailed keys and values.
+ *
+ * <p>If the specified Paimon table does not exist, this action will automatically create the table.
+ * Its schema will be derived from all specified Kafka topic. If the Paimon table already exists,
+ * its schema will be compared against the schema of all specified Kafka topic.
+ *
+ * <p>This action supports a limited number of schema changes. Unsupported schema changes will be
+ * ignored. Currently supported schema changes includes:
+ *
+ * <ul>
+ *   <li>Adding columns.
+ *   <li>Altering column types. More specifically,
+ *       <ul>
+ *         <li>altering from a string type (char, varchar, text) to another string type with longer
+ *             length,
+ *         <li>altering from a binary type (binary, varbinary, blob) to another binary type with
+ *             longer length,
+ *         <li>altering from an integer type (tinyint, smallint, int, bigint) to another integer
+ *             type with wider range,
+ *         <li>altering from a floating-point type (float, double) to another floating-point type
+ *             with wider range,
+ *       </ul>
+ *       are supported. Other type changes will cause exceptions.
+ * </ul>
+ */
+public class KafkaSyncTableAction implements Action {
```

Comment:
We can create a PR for `KafkaSyncTableAction` now~

Reply:
https://github.com/apache/incubator-paimon/pull/1239